src/C-DEngine/C-DCommunication/cdeQueuedSender/cde_QueuedSenderREST.cs

```diff
@@ -259,8 +272,14 @@ public void TheRESTSenderThread()
                         throw new Exception("Client Certificate could not be added");
                     }
                     MyREST.PostRESTAsync(pData, sinkUploadDataCompleted, FireSenderProblem);
-                    if (BinSendBuffer != null)
+                    if (BinSendBuffer is {Length: > 0})
```

Comment:
and here: if (EnableKPIs && BinSendBuffer...

Reply:
Please see the new changes in `TheCDEKpis`.